modules/apps/headless/headless-delivery/headless-delivery-test/src/testFunctional/tests/SiteDocumentEntry.testcase

```diff
@@ -240,4 +278,108 @@ definition {
+			var filePath = TestCase.getDependenciesDirPath(fileName = "Document_1.jpg");
+
+			var response = DocumentAPI.addDocumentWithFileInSite(
+				dmDocumentTitle = "DM Document Title",
+				fileName = "FileName.jpg",
+				filePath = ${filePath},
+				siteId = ${siteId});
+		}
+
+		task ("When a put request is made to update a document with an invalid file name") {
+			var documentId = JSONPathUtil.getIdValue(response = ${response});
+
+			var response = DocumentAPI.updateDocumentByDocumentId(
+				documentId = ${documentId},
+				fileName = "FileNameEdit?.jpg",
+				title = "DM Document Title Edit");
+
+			echo("####After updating the document: '${response}'");
+		}
+
+		task ("Then file name and title can not be updated") {
+			TestUtils.assertPartialEquals(
+				actual = ${response},
+				expected = "BAD_REQUEST");
+
+			TestUtils.assertPartialEquals(
+				actual = ${response},
+				expected = "Invalid file name FileNameEdit?.jpg");
+
+			User.loginPG(
+				password = "test",
+				userEmailAddress = "test@liferay.com");
+
+			DMNavigator.openToEntryInAdmin(
+				dmDocumentTitle = "DM Document Title",
+				groupName = "Guest",
+				siteURLKey = "guest");
+
+			DMDocument.viewCP(
+				dmDocumentFileName = "FileName.jpg",
+				dmDocumentTitle = "DM Document Title");
+		}
+	}
+
+	@priority = 5
+	test CanUpdateDocumentFileName {
+		task ("Given a document in a site") {
+			var siteId = JSONGroupAPI._getGroupIdByNameNoSelenium(
+				groupName = "Guest",
+				site = "true");
+			var filePath = TestCase.getDependenciesDirPath(fileName = "Document_1.jpg");
+
+			var response = DocumentAPI.addDocumentWithFileInSite(
+				dmDocumentTitle = "DM Document Title",
+				fileName = "FileName.jpg",
+				filePath = ${filePath},
+				siteId = ${siteId});
+		}
+
+		task ("When a put request is made to update document file name and title") {
+			var documentId = JSONPathUtil.getIdValue(response = ${response});
+
+			var response = DocumentAPI.updateDocumentByDocumentId(
+				documentId = ${documentId},
+				fileName = "FileNameEdit.jpg",
+				title = "DM Document Title Edit");
+
+			echo("####After updating the document: '${response}'");
+		}
+
+		task ("Then I receive a correct body response with updated file name and title") {
+			var documentFileName = DocumentAPI.getFileNameOfDocument(responseToParse = ${response});
+
+			TestUtils.assertEquals(
+				actual = ${documentFileName},
+				expected = "FileNameEdit.jpg");
+
+			var documentTitle = DocumentAPI.getTitleOfDocument(responseToParse = ${response});
+
+			TestUtils.assertEquals(
+				actual = ${documentTitle},
+				expected = "DM Document Title Edit");
+
+			User.loginPG(
+				password = "test",
+				userEmailAddress = "test@liferay.com");
+
+			DMNavigator.openToEntryInAdmin(
+				dmDocumentTitle = "DM Document Title Edit",
+				groupName = "Guest",
+				siteURLKey = "guest");
+
+			DMDocument.viewCP(
+				dmDocumentFileName = "FileNameEdit.jpg",
+				dmDocumentTitle = "DM Document Title Edit");
```

Comment:
Let's skip this UI assertion as well to ensure headless tests are all very stable

Reply:
Updated, thanks!